src/System.Globalization/tests/NumberFormatInfo/NumberFormatInfoGetInstance.cs

```diff
@@ -24,7 +24,38 @@ public static IEnumerable<object[]> GetInstance_TestData()
         [MemberData(nameof(GetInstance_TestData))]
         public void GetInstance(IFormatProvider formatProvider, NumberFormatInfo expected)
         {
-            Assert.Equal(expected, NumberFormatInfo.GetInstance(formatProvider));
+            NumberFormatInfo nfi = NumberFormatInfo.GetInstance(formatProvider);
+            if (nfi.Equals(expected))
+            {
+                return;
+            }
```

Comment:
Nit: Seems like the only reason to do this check now is as a perf optimization, and it seems unlikely that's an issue for the test.  Maybe we should just remove this Equals check and always do the ones below?